### ISSUE TITLE
semicolon rule false positives with class members with arrow functions

### ISSUE THREAD

**Issue by HamletDRC** [palantir/tslint#1895]
### Bug Report

- __TSLint version__: `4.0.2`
- __TypeScript version__: `2.1.4`
- __Running TSLint via__: cli and grunt-tslint

#### TypeScript code being linted

```ts
/**
 */
export class Rule {

    public FAILURE_STRING_FACTORY = () => {
        return;
    };
}
```

with `tslint.json` configuration:

```json
    "semicolon": [true],

```

#### Actual behavior

unnecessary semicolon is reported. 
#### Expected behavior

the semicolon after the initializer expression is not unnecessary if you are requiring semi-colons.

**Comment by nchen63** [comment]
see https://github.com/palantir/tslint/issues/1476. This was intentional, and you can get the old behavior with the option `ignore-bound-class-methods`
